src/DeferredRender.js

```diff
@@ -0,0 +1,32 @@
+/* eslint react/prop-types: 0 */
+// @flow
+import * as React from 'react';
+
+let rIC;
+
+if (window.requestIdleCallback) {
+  rIC = window.requestIdleCallback;
+} else {
+  rIC = (handler, { timeout }) => setTimeout(handler, timeout);
```

Comment:
The timeout is the maximum delay with requestIdleCallback, so it always taking the maximum time in the fallback seems bad (e.g. if rIC is used with a 1000ms timeout, it might execute the code 1ms in the future, making the fallback 1000x slower).

For the fallback, I would just do a two-frame delay with requestAnimationFrame to let other things render and paint before we try to render the stuff.

Basically it would look like this:

```js
riC = (handler) => requestAnimationFrame(() => requestAnimationFrame(handler));
```

non-basically, we need to handle canceling both the real rIC and the current requestAnimationFrame call when the component unmounts, so we should probably define both variants of the rIC variable to return a cancelation function (similar to CtxStore's subscriptions).